src/files/pyfingerprint/pyfingerprint.py

```diff
@@ -998,23 +998,25 @@ def storeTemplate(self, positionNumber = -1, charBufferNumber = FINGERPRINT_CHAR
         else:
             raise Exception('Unknown error '+ hex(receivedPacketPayload[0]))
 
-    def searchTemplate(self):
+    def searchTemplate(self, charBufferNumber = FINGERPRINT_CHARBUFFER1, positionStart = 0, count = -1):
         """
         Search the finger characteristics in CharBuffer in database.
 
+        @param integer(1 byte) charBufferNumber
+        @param integer(2 byte) positionStart
+        @param integer(2 byte) count
+
         Return a tuple that contain the following information:
         0: integer(2 bytes) The position number of found template.
         1: integer(2 bytes) The accuracy score of found template.
 
         @return tuple
         """
 
-        ## CharBuffer1 and CharBuffer2 are the same in this case
-        charBufferNumber = FINGERPRINT_CHARBUFFER1
-
-        ## Begin search at index 0
-        positionStart = 0x0000
-        templatesCount = self.getStorageCapacity()
+        if ( count < 0 ):
```

Comment:
Please invert the cases, i prefer `count > 0` and directly in the first case use `count` to improve readability. And btw: And we should remove the spaces in the braces (they does not match the official Python code style imho)

Reply:
Yes, I agree. We should do a refactoring to remove the unnecessary blanks.